### PR TITLE
Relax frontend config validation and prune public ingress when public IP is not present

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

Application Gateway is now testing private-ip frontend only feature. Currently, AGIC has a fatal validation that requires AppGW to have a public IP.

This PR
* removes the frontend IP validation on AGIC
* prunes ingresses that need to be attached to public when public IP is not present on the Gateway.

Testing:

### Gateway has private IP only
image

* When ingress is missing annotation, then event is emitted.
image

* When ingress has annotation, then it correctly programmed.
  image
  * Listener uses private IP
image
  * Routing Rule attached to private IP listener
image
  * Backend Pool has the referenced service  
  image

### CODE REVIEW

File: functional_tests/private-ip-only-gateway.json [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            }
+        ],
+        "httpListeners": [
+            {
+                "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/httpListeners/fl-08ba2bb7da9df5927d900fca8ce96ba5",
+                "name": "fl-08ba2bb7da9df5927d900fca8ce96ba5",
+                "properties": {
+                    "frontendIPConfiguration": {
+                        "id": "--front-end-ip-id-2--"
+                    },
```
Comment (akshaysngupta) [inline functional_tests/private-ip-only-gateway.json:91]: listener is attached to private IP frontend

File: functional_tests/private-ip-only-gateway.json [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                    }
+                }
+            }
+        ],
+        "requestRoutingRules": [
+            {
+                "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/requestRoutingRules/rr-08ba2bb7da9df5927d900fca8ce96ba5",
+                "name": "rr-08ba2bb7da9df5927d900fca8ce96ba5",
+                "properties": {
+                    "httpListener": {
```
Comment (akshaysngupta) [inline functional_tests/private-ip-only-gateway.json:189]: routing rule is correctly attached to the listener.

File: functional_tests/private-ip-only-gateway.json [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+                    "probe": {
+                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/probes/defaultprobe-Http"
+                    },
+                    "protocol": "Http",
+                    "requestTimeout": 30
+                }
+            }
+        ],
+        "frontendIPConfigurations": [
+            {
```
Comment (akshaysngupta) [inline functional_tests/private-ip-only-gateway.json:60]: only private ip present on the gateway

**Review by alxyzc — APPROVED** [review]
LGTM!

**Pull request merged**